// @nestjs-addons/in-memory-db/dist/v1/in-memory-db.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var InMemoryDBV1Module_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.InMemoryDBV1Module = void 0;
const common_1 = require("@nestjs/common");
const services_1 = require("./services");
const providers_1 = require("./providers");
let InMemoryDBV1Module = InMemoryDBV1Module_1 = class InMemoryDBV1Module {
    static forRoot(config = {}) {
        const providers = providers_1.createInMemoryDBV1ForRootProviders(config);
        return {
            module: InMemoryDBV1Module_1,
            providers,
            exports: providers,
        };
    }
    static forFeature(featureName, config = {}) {
        const providers = providers_1.createInMemoryDBV1ForFeatureProviders(featureName, config);
        return {
            module: InMemoryDBV1Module_1,
            providers,
            exports: providers,
        };
    }
};
InMemoryDBV1Module = InMemoryDBV1Module_1 = __decorate([
    common_1.Module({
        providers: [services_1.InMemoryDBV1Service],
        exports: [services_1.InMemoryDBV1Service],
    })
], InMemoryDBV1Module);
exports.InMemoryDBV1Module = InMemoryDBV1Module;
