// @nestjs-addons/in-memory-db/dist/in-memory-db.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var InMemoryDBModule_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.InMemoryDBModule = void 0;
const common_1 = require("@nestjs/common");
const services_1 = require("./services");
const providers_1 = require("./providers");
let InMemoryDBModule = InMemoryDBModule_1 = class InMemoryDBModule {
    static forRoot(config = {}) {
        const providers = providers_1.createInMemoryDBForRootProviders(config);
        return {
            module: InMemoryDBModule_1,
            providers,
            exports: providers,
        };
    }
    static forFeature(featureName, config = {}) {
        const providers = providers_1.createInMemoryDBForFeatureProviders(featureName, config);
        return {
            module: InMemoryDBModule_1,
            providers,
            exports: providers,
        };
    }
};
InMemoryDBModule = InMemoryDBModule_1 = __decorate([
    common_1.Module({
        providers: [services_1.InMemoryDBService],
        exports: [services_1.InMemoryDBService],
    })
], InMemoryDBModule);
exports.InMemoryDBModule = InMemoryDBModule;
